packages/helm-charts/oracle-rbac/templates/role.yaml

```diff
@@ -1,8 +1,12 @@
+{{ range $index, $e := until (.Values.oracleCount | int) }}
 apiVersion: rbac.authorization.k8s.io/v1
 kind: Role
 metadata:
-  name: {{ template "name" . }}
+  name: {{ template "name" (dict "Values" $.Values "index" $index) }}
```

Comment:
You should be able to declare variables in helm (see https://helm.sh/docs/chart_template_guide/variables/#:~:text=In%20Helm%20templates%2C%20a%20variable,a%20variable%20for%20Release.Name%20.), what do you think in each file of creating a variable like
```
{{- $some_smart_variable_name := (dict "Values" $.Values "index" $index) -}}
```
So we can just do

```
{{ template "name" $some_smart_variable_name }}
```